src/osd/PrimaryLogPG.cc

```diff
@@ -5992,6 +5992,70 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 
       break;
 
+    case CEPH_OSD_OP_SET_CHUNK:
```

Comment:
This operation doesn't make much sense to me.  AFAICS,

- it uses the same target object name for all chunks.  When would that be useful?
- the op always starts at offset 0, and 
- the chunks have to be the same size

Since we can already put multiple set_chunk ops in a single reqeuest, it seems like this op should take

- offset, length
- target oid
- target offset

and shouldn't need to loop.  (I can't think of a reason why we would want a bunch of fixed-size consecutive chunks pointing at the same object?)

Reply:
Fixed.  It seems that I misunderstood "set_chunk" op. I have assumed that set_chunk op update all of the chunk_map entries and flush the object. (the object in the base pool is flushed to the target pool).